crates/router/src/core/files/transformers.rs

```diff
@@ -0,0 +1,257 @@
+                create_file_request.file_type.clone(),
+            );
+            match validation {
+                Ok(()) => Ok(()),
+                Err(err) => match err.current_context() {
+                    errors::ConnectorError::FileValidationFailed { reason } => {
+                        Err(errors::ApiErrorResponse::FileValidationFailed {
+                            reason: reason.to_string(),
+                        }
+                        .into())
+                    }
+                    _ => Err(errors::ApiErrorResponse::InternalServerError.into()),
+                },
+            }
+        }
+    }
+}
+
+pub async fn delete_file_using_file_id(
+    state: &AppState,
+    file_key: String,
+    merchant_account: &storage_models::merchant_account::MerchantAccount,
+) -> CustomResult<(), errors::ApiErrorResponse> {
+    let file_metadata_object = state
+        .store
+        .find_file_metadata_by_merchant_id_file_id(&merchant_account.merchant_id, &file_key)
+        .await
+        .change_context(errors::ApiErrorResponse::FileNotFound)?;
+    match file_metadata_object.file_upload_provider {
+        storage_models::enums::FileUploadProvider::Hyperswitch => {
+            delete_file(
+                #[cfg(feature = "s3")]
+                state,
+                file_metadata_object.provider_file_id,
+            )
+            .await
+        }
+        _ => Err(errors::ApiErrorResponse::NotImplemented {
+            message: errors::api_error_response::NotImplementedMessage::Default,
+        }
+        .into()),
+    }
+}
+
+pub async fn retrieve_file_and_provider_file_id_from_file_id(
+    state: &AppState,
+    file_id: Option<String>,
+    merchant_account: &storage_models::merchant_account::MerchantAccount,
+) -> CustomResult<(Option<Vec<u8>>, Option<String>), errors::ApiErrorResponse> {
+    match file_id {
+        None => Ok((None, None)),
+        Some(file_key) => {
+            let file_metadata_object = state
+                .store
+                .find_file_metadata_by_merchant_id_file_id(&merchant_account.merchant_id, &file_key)
+                .await
+                .change_context(errors::ApiErrorResponse::FileNotFound)?;
+            match file_metadata_object.file_upload_provider {
+                storage_models::enums::FileUploadProvider::Hyperswitch => Ok((
+                    Some(
+                        retrieve_file(
+                            #[cfg(feature = "s3")]
+                            state,
+                            file_metadata_object.provider_file_id.clone(),
+                        )
+                        .await?,
+                    ),
+                    Some(file_metadata_object.provider_file_id),
+                )),
+                //TODO: Handle Retrieve for other providers
+                _ => Ok((None, Some(file_metadata_object.provider_file_id))),
+            }
+        }
+    }
+}
+
+pub async fn upload_and_get_provider_provider_file_id(
```

Comment:
is the two provider intentional? seems like a typo.

Reply:
It's intentional as we return both `provider` and `provider_file_id`